14_contains/solution/contains-solution.spec.js

```diff
@@ -0,0 +1,55 @@
+const contains = require('./contains-solution');
+
+describe('contains', () => {
+  const object = {
```

Comment:
I think it could also be nice to add a case that makes sure the algorithm can properly handle sibling branches and isn't all just children. Can maybe just reorganize the object here to cover this.

You can see an example of what I mean by looking at the original solution on the [site currently linked in the curriculum](https://www.codingame.com/playgrounds/5422/js-interview-prep-recursion). Their solution is:
```js
function contains(obj, value) {
  for(var key in obj){
  if(typeof obj[key] === 'object'){
      return contains(obj[key], value);
    }

    if (obj[key] === value){
      return true;
    }
  }
  return false;
}
```
If you pass a nested object that's just all children, it's fine
```js
const object = {
  data: {
    childOne: {
      foo: "bar",
      childTwo: {
        number: 42
      }
    }
  }
}

// true
console.log(contains(object, 42));
```

But if you pass it an object where there are "sibling" branches (ie nested objects on the same level), you'll see there's a logical flaw.

```js
const objectWithSiblings = {
  data: {
    childOne: {
      foo: "bar",
    },
    childTwo: {
      number: 42
    }
  }
}

// outputs `false` but should be `true`!
console.log(contains(objectWithSiblings, 42)
```

Your solution naturally handles this just fine, but I think the tests should look for it because it's a common problem people can run into that would sneak through currently.

Reply:
Nicely spotted! My solution did not actually work with this new requirement, so I had to spend some time thinking how to fix it. Its been a while since I used recursion or queues so this was certainly a nice refresh